contracts/vortex/CarbonVortex.sol

```diff
@@ -183,6 +183,28 @@ contract CarbonVortex is ICarbonVortex, Upgradeable, ReentrancyGuardUpgradeable,
         _allocateRewards(msg.sender, tokens, rewardAmounts);
     }
 
+    /**
+     * @dev withdraws funds held by the contract and sends them to an account
+     *
+     * requirements:
+     *
+     * - the caller must be the admin of the contract
+     */
+    function withdrawFunds(
+        Token token,
+        address payable target,
+        uint256 amount
+    ) external validAddress(target) nonReentrant onlyAdmin {
```

Comment:
`onlyAdmin` by itself takes care of reentrancy, making `nonReentrant` somewhat redundant (in terms of gas).

Reply:
It’s a good practice, gas is irrelevant here

Reply:
But there is no real reentrancy concern here anyway, but as @yudilevi said - better safe then sorry